### PR TITLE
Add Blazor topics

### LINKED ISSUE #10825: Add Blazor topics
React to: https://github.com/aspnet/Docs/pull/10800#discussion_r255284040

- [x] **Title:** *Introduction to Blazor* &ndash; **Location:** aspnetcore/client-side/spa/blazor/index.md
- [x] **Title:** *Get started with Blazor* &ndash; **Location:** aspnetcore/client-side/spa/blazor/get-started.md
- [x] Refactor *Build your first Razor Components app* to include Blazor. Provide a separate downloadable sample app for Blazor.

### PR DESCRIPTION
**Pull request by guardrex**
Fixes #10825 

Internal Review Topics:

* [Blazor Overview](https://review.docs.microsoft.com/en-us/aspnet/core/client-side/spa/blazor/index?view=aspnetcore-3.0&branch=pr-en-us-10873)
* [Blazor: Get started](https://review.docs.microsoft.com/en-us/aspnet/core/client-side/spa/blazor/get-started?view=aspnetcore-3.0&branch=pr-en-us-10873)
* [Razor Components Overview](https://review.docs.microsoft.com/en-us/aspnet/core/razor-components/index?view=aspnetcore-3.0&branch=pr-en-us-10873)
* [Razor Components: Get started](https://review.docs.microsoft.com/en-us/aspnet/core/razor-components/get-started?view=aspnetcore-3.0&branch=pr-en-us-10873)
* [Build your first app](https://review.docs.microsoft.com/en-us/aspnet/core/tutorials/build-your-first-razor-components-app?view=aspnetcore-3.0&branch=pr-en-us-10873)

Notes:

* Can't name both RC and Blazor samps with the same name, so I went with "Razor Components" and "Blazor" as the app names ... super-obvious that way when readers look in the tutorial's *samples* folder.
* Still needs work on the intro paragraphs of each of the introduction (*Overview*/*index.md*) topics. Blazor intro needs the list of main features. If you need to make big layout changes, I recommend that you edit these directly here, then I'll do a style+grammar pass afterward.
* Made sense to jettison the "hosting models" sections now that these intro topics are entirely dedicated to either RC or Blazor. I retain a brief remark at the end of each about the converse approach (RC mentions Blazor and Blazor mentions RC) with a cross-link.
* Dan, your recommendations worked well. I 🖖 u on those calls.

### CODE REVIEW

File: aspnetcore/razor-components/index.md [hunk 1]
```diff
@@ -20,35 +20,7 @@ By [Daniel Roth](https://github.com/danroth27) and [Luke Latham](https://github.
 * Share server-side and client-side app logic all written with .NET.
```
Comment (danroth27) [inline aspnetcore/razor-components/index.md]: I think we should remove the include at the beginning of this doc that adds the note about Blazor being experimental. It's not really relevant for this topic.

File: aspnetcore/tutorials/build-your-first-razor-components-app.md [hunk 2]
```diff
@@ -143,7 +153,7 @@ Add a new page to the app that implements a simple todo list.
 
 1. Add a *TodoItem.cs* file to the root of the project to hold a class that represents a todo item. Use the following C# code for the `TodoItem` class:
 
-   [!code-cshtml[](build-your-first-razor-components-app/samples/3.x/WebApplication1/WebApplication1.App/TodoItem.cs)]
+   [!code-cshtml[](build-your-first-razor-components-app/samples/3.x/RazorComponents/RazorComponents.App/TodoItem.cs)]
 
 1. Return to the Todo component (*Todo.cshtml*):
```
Comment (danroth27) [inline aspnetcore/tutorials/build-your-first-razor-components-app.md]: The highlighting in the code snippet for this section looks a bit off.

File: aspnetcore/client-side/spa/blazor/get-started.md [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+ms.author: riande
+ms.custom: mvc
+ms.date: 02/11/2019
+uid: spa/blazor/get-started
+---
+# Get started with Blazor
+
+By [Daniel Roth](https://github.com/danroth27) and [Luke Latham](https://github.com/guardrex)
+
+[!INCLUDE[](~/includes/razor-components-preview-notice.md)]
```
[the remark this answers is not in the document]
Comment (guardrex) [reply aspnetcore/client-side/spa/blazor/get-started.md]: Cool ... I'll make this update on the PR this afternoon that also addresses the Blazor TOC additions. I'll do it right after we merge this one.

File: aspnetcore/tutorials/build-your-first-razor-components-app.md (hunk 2 above)
Comment (guardrex) [reply aspnetcore/tutorials/build-your-first-razor-components-app.md]: I double-checked locally, and it looks :eye: ok. I can't see the doc's system build (internal review topic), but I'll find out quickly right after it goes live what's what. I can do a quick patch if anything breaks. Locally, these highlights seem to be ok.

**Comment by guardrex** [comment]
@tdykstra You can make it a quickie look (I'd like to merge and get right back into these topics with an updated master) ... we'll be back here for Preview 3 making a final pass. I'm mostly concerned about your approval of the TOC bits.

@scottaddie We'll have time for u to :eye: it for Preview 3 ..... you should be dug out ⛄️ by then I hope.

**Comment by danroth27** [comment]
@tdykstra These TOC updates sound fine to me, although I would prefer to have Razor Components in first position, and then SPA second under client-side dev.

**Comment by guardrex** [comment]
> A bunch of TOC nodes under Razor Components 404 -- is that expected at this point?

I'm not getting that here. No RC topics 404 here. Which ones are 404'ing there?

> delete Overview under Single Page Apps (it 404s)

It loads here as https://docs.microsoft.com/aspnet/core/client-side/spa/

I'll work a commit along the suggestions provided.

**Comment by guardrex** [comment]
@tdykstra @danroth27 

* Pointed the Client-side Overview to the Razor Components Overview. You said Blazor, but RC is the supported tech at this point. Blazor is the first link tho under SPA. I perhaps guessed wrong, so let me know.
* Moved RC to the top of the node. 
* Moved LibMan under RC.

**Comment by tdykstra** [comment]
>No RC topics 404 here. Which ones are 404'ing there?

They're working now.

>It loads here as https://docs.microsoft.com/aspnet/core/client-side/spa/

That one works now too.  It's one of those duplicate-the-toc files, how about delete and redirect to Blazor since we don't have a real SPA overview?

**Comment by guardrex** [comment]
@tdykstra Done on the last commit ... see how that looks.

**Comment by guardrex** [comment]
... but I only deleted. If you still want "Overview" but linked to the Blazor Overview, I still need to do that.

**Comment by guardrex** [comment]
Feel free to edit directly, of course for what you want.

**Comment by tdykstra** [comment]
OK, editing TOC now, a couple of tweaks.  Also, the /aspnet/core/client-side/spa/index shows the page now but is disconnected from the TOC.  Need to redirect the URL to Blazor

**Comment by guardrex** [comment]
Yes, :+1: redirect entry. Let me know if u want me to add that or if you're going to do it.

**Comment by guardrex** [comment]
@tdykstra Try that redirect ... see if that looks good. :point_up:

**Comment by tdykstra** [comment]
Thanks! -- if the TOC changes build OK I think we're good.

**Comment by guardrex** [comment]
Looks like we need a client-side/index redirect, too. I'll add that next.

**Comment by guardrex** [comment]
That might&dagger; work.

**&dagger;might** ... if the :pray: *server gods* :pray: are happy with us.

**Comment by guardrex** [comment]
@tdykstra Look at it this way .......

...... if this all goes to heck in a handbasket ........

**I hear Apple docs _ARE HIRING!_** 😀😀😀😀🙈:smile:

@danroth27 ... look good to you? ... I'll get this merged and on to the Blazor TOC/INCLUDE updates next.

**Comment by guardrex** [comment]
It wasn't too pleased with the file left on the repo, so I 🔪 the file on the last commit.

**Comment by scottaddie** [comment]
I agree that LibMan should appear in the ToC before Gulp and Grunt. We do own that product, after all.

**Comment by danroth27** [comment]
:shipit:

**Pull request merged**